ipatests/test_integration/test_ipa_cert_fix.py

```diff
@@ -278,52 +292,86 @@ def install(cls, mh):
+        self.replicas[0].run_command(['ipa', 'user-show', testuser])
+
+        # move date back on replica and master
+        for host in self.master, self.replicas[0]:
+            move_date(host, 'start', '-3years-1days')
+
+    def test_renew_expired_cert_replica(self, move_date_check_replication):
         """Test renewal of certificates on replica with ipa-cert-fix
 
         This is to check that ipa-cert-fix renews the certificates
         on replica
 
         related: https://pagure.io/freeipa/issue/7885
         """
-        move_date(self.master, 'stop', '+3years+1days')
-
         # wait for cert expiry
         check_status(self.master, 8, "CA_UNREACHABLE")
 
         self.master.run_command(['ipa-cert-fix', '-v'], stdin_text='yes\n')
 
         check_status(self.master, 9, "MONITORING")
 
-        # move system date to expire cert on replica
-        move_date(self.replicas[0], 'stop', '+3years+1days')
-
+        # replica operations
         # RA agent cert will be expired and in CA_UNREACHABLE state
         check_status(self.replicas[0], 1, "CA_UNREACHABLE")
 
+        # check that HTTP,LDAP,PKINIT are renewd and in MONITORING state
+        for cert in ['http', 'dirsrv', 'kdc']:
+            req_ids = get_req_id(self.replicas[0], cert, '3')
+            tasks.wait_for_certmonger_status(
+                self.replicas[0], ('MONITORING'), req_ids[0], timeout=600
+            )
+
+        # restart ipa services after cert renewal
+        self.replicas[0].run_command(
+            ['ipactl', 'restart', '--ignore-service-failures']
+        )
         # renew RA agent cert
         self.replicas[0].run_command(
             ['ipa-cert-fix', '-v'], stdin_text='yes\n'
         )
 
-        # LDAP/HTTP/PKINIT certs will be renewed automaticaly
-        # after moving date on replica. This 3, 1 CA cert,
-        # 1 RA agent cert. Check for total 5 valid certs.
-        check_status(self.replicas[0], 5, "MONITORING")
+        req_ids = get_req_id(
+            self.replicas[0], '"Server-Cert cert-pki-ca"', '4'
+        )
+        tasks.wait_for_certmonger_status(
+            self.replicas[0], ('MONITORING'), req_ids[0], timeout=600
+        )
 
         # get the req ids of all certs to renew remaining
         # certs by re-submitting it
-        result = self.replicas[0].run_command(['getcert', 'list'])
-        req_ids = re.findall(r'\d{14}', result.stdout_text)
+        req_ids = get_req_id(self.replicas[0], 'CA_WORKING', '1')
 
         # resubmit the certs to renew them
-        for req_id in req_ids:
-            self.replicas[0].run_command(
-                ['getcert', 'resubmit', '-i', req_id]
-            )
-
-        check_status(self.master, 9, "MONITORING")
-
-        # move date back on replica and master
-        move_date(self.replicas[0], 'start', '-3years-1days')
-        move_date(self.master, 'start', '-3years-1days')
+        if req_ids:
+            for req_id in req_ids:
```

Comment:
Since re.findall returns an empty list if no match is found, the 2 lines can be simplified by removing the `if req_ids:`.